CryptoPrices/Sources/App/AppCoordinator.swift

```diff
@@ -0,0 +1,48 @@
+//
+//  AppCoordinator.swift
+//  CryptoPrices
+//
+//  Created by Khanh on 19/12/2022.
+//
+
+import Combine
+import Home
+import MyCoin
+import SwiftUI
+
+final class AppCoordinator: ObservableObject {
+
+    @Published var state: AppState = .home(HomeState())
+
+    private var cancellables = Set<AnyCancellable>()
+
+    init() {
+        $state
+            .compactMap { currentState -> HomeState? in
+                if case let .home(homeState) = currentState {
+                    return homeState
+                }
+                return nil
+            }
+            .flatMap(\.$didSelectCoin)
+            .filter { $0 }
+            .sink(receiveValue: { [weak self] _ in
+                self?.state = .myCoin(MyCoinState())
+            })
+            .store(in: &cancellables)
+
+        $state
+            .compactMap { currentState -> MyCoinState? in
+                if case let .myCoin(myCoinState) = currentState {
+                    return myCoinState
+                }
+                return nil
+            }
+            .flatMap(\.$didSelectBack)
+            .filter { $0 }
+            .sink(receiveValue: { [weak self] _ in
+                self?.state = .home(HomeState())
+            })
+            .store(in: &cancellables)
```

Comment:
I am wondering about the scalability of this implementation as for now there is just a simple navigation between 2 screens and we need to handle a lot of navigation code already. Thus, it looks like there is going to be a lot of implementations for this state in this `AppCoordinator` file the more screens we have. 😅

What do you guys think about this [approach](https://quickbirdstudios.com/blog/coordinator-pattern-in-swiftui/) that we involves the coordinator directly into each View's viewModel instead of encapsulating all the states at the App level like the current implementation? This allows better decoupling and make our individual scenes completely independent from their view context. 🙏

Reply:
Thanks for the raise @minhnimble. As Thieu mentioned above, with this navigation we could have a couple of approaches but to satisfy the Home and MyCoin shouldn't know each other and they are in the horizontal layer as we discussed [at here](https://nimble-co.slack.com/archives/C04C7E5PWUQ/p1671424415289429). Therefore, we go with this solution which is the most suitable now 🙏 .

Reply:
Thank you for your insights @minhnimble 🙏
Totally agree with you!  This is far from a finished solution. And not only us, developers from the community still struggle to find the best solution for this problem, in both UIKit and SwiftUI. I still think we have to address this sooner or later, with a simple, easy-to-implement approach!